src/main/java/sysc3303/a1/group3/Parser.java

```diff
@@ -71,4 +71,58 @@ private Event parseIncidentLine(String line) throws IllegalArgumentException {
             throw new IllegalArgumentException("Invalid line: " + line);
         }
     }
+
+
+
+    public ArrayList<Zone> parseZoneFileLine(InputStream file) {
+        ArrayList<Zone> zones = new ArrayList<>();
+
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(file))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                try {
+                    zones.add(parseZoneFileLine(line));
+                } catch (IllegalArgumentException e) {
+                    System.err.printf("Error parsing line: %s with exception: %s, continuing to next line.%n", line, e);
+                    continue;
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading Zone file with exception: " + e);
+        }
```

Comment:
I would suggest letting this method throw the IOException so that the caller can properly handle the exception